src/intrinsics/ecma262/WeakMap.js

```diff
@@ -40,6 +40,12 @@ export default function(realm: Realm): NativeFunctionValue {
     map.$WeakMapData = [];
 
     // 4. If iterable is not present, let iterable be undefined.
+    if (iterable && realm.isCompatibleWith(realm.MOBILE_JSC_VERSION)) {
+      throw realm.createErrorThrowCompletion(
+        realm.intrinsics.TypeError,
+        "the weak map constructor doesn't take arguments"
+      );
+    }
```

Comment:
Does JSC in fact throw a runtime type error in this situation? If not, it may be preferable to emit a compile time diagnostic here.